### PR TITLE
Add GremlinSource to GremlinEntityInformation

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Align the constructor parameters to entityinformation.

* Add new field gremlinSource to entityInformation.

Signed-off-by: Pan Li <incarnation.p.lee@outlook.com>

### CODE REVIEW

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSourceWriter.java [hunk 1]
```diff
@@ -5,23 +5,24 @@
...
 
-    public BasicGremlinSourceWriter(@NonNull Object domain) {
-        this.entityInformation = new GremlinEntityInformation(domain);
+    public BasicGremlinSourceWriter(@NonNull Class<?> domainClass) {
+        this.entityInformation = new GremlinEntityInformation(domainClass);
     }
 
     protected String getPersistentEntityId() {
-        return entityInfo.getIdField().toString();
+        return this.entityInformation.getIdField().toString();
```
Comment (yungezz) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSourceWriter.java:21]: what if not specific id field?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinSourceEdgeWriter.java [hunk 2]
```diff
@@ -18,16 +19,17 @@
...
-        Assert.notNull(domain, "domain class should not be null");
+    @SuppressWarnings("unchecked")
+    private String getPersistentEntityId(@NonNull Class<?> domainClass) {
+        Assert.notNull(domainClass, "domain class should not be null");
 
-        final GremlinEntityInformation entityInformation = new GremlinEntityInformation(domain);
+        final GremlinEntityInformation entityInformation = new GremlinEntityInformation(domainClass);
 
-        return entityInfo.getIdField().toString();
+        return entityInformation.getIdField().toString();
```
Comment (yungezz) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinSourceEdgeWriter.java:32]: any null handling if not specify id?

File: src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java [hunk 3]
```diff
@@ -23,19 +25,29 @@
     private Field id;
     private String entityLabel;
     private GremlinEntityType entityType;
+    private GremlinSource gremlinSource;
 
-    public GremlinEntityInformation(Class<T> domainClass) {
+    public GremlinEntityInformation(@NonNull Class<T> domainClass) {
```
Comment (yungezz) [inline src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java:30]: seems there's a `@notempty` annotation used in azure-spring-boot, you may want to be consistent

File: src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java [hunk 4]
```diff
@@ -64,7 +76,7 @@ public ID getId(T entity) {
         return (Class<ID>) this.id.getType();
     }
 
-    private Field getIdField(Class<?> domainClass) {
+    private Field getIdField(@NonNull Class<?> domainClass) {
```
Comment (yungezz) [inline src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java:79]: what's return value if user not specify id field?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSourceWriter.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/BasicGremlinSourceWriter.java:21]: GremlinEntityInformation will take care of domainClass in constructor, if we cannot figure out the only one id Field. One runtime exception will be raised.

File: src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/repository/support/GremlinEntityInformation.java:30]: `@NotEmpty` is for String, Array, map and Collection, include null and not empty. May not very suitable here.

**Review by yungezz — APPROVED** [review]

**Pull request merged**